File: front/src/model/UserDTO.ts
import {Role} from "@/enum/UserRole";
import {LocationDTO} from "@/model/LocationDTO";

export interface UserDTO {
    id?: number,
    username: string,
    firstname: string,
    middlename: string,
    lastname: string,
    roles: Role[],
    locationIds: number[],
    locations: LocationDTO[],
    password: string,
    isActive: boolean
}

File: src/main/java/ru/didcvee/ecd/user/dto/UserDto.java
package ru.didcvee.ecd.user.dto;

import lombok.Data;

import java.util.List;

@Data
public class UserDto {
    private Long id;
    private String username;
    private String firstname;
    private String middlename;
    private String lastname;
    private Boolean isActive;
    private List<String> roles;
    private List<Long> subjectIds;
    private List<SubjectDto> subjects;
    private Long groupId;
}
